Tools/Taxonomy/OwaspMobileTop10/OwaspMobileTop10TaxonomyGenerator.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+using Markdig.Syntax;
+
+using Microsoft.CodeAnalysis.Sarif;
+
+using Newtonsoft.Json;
+
+using Taxonomy.Common;
+
+namespace Taxonomy
+{
+    public class OwaspMobileTop10TaxonomyGenerator : TaxonomyGenerator
+    {
+        public bool SaveToSarif(string sourceFolderPath, string targetFilePath, string version)
+        {
+            try
+            {
+                List<OwaspMobileTop10Record> results = this.ReadFromMd(sourceFolderPath);
+
+                Run run = this.ConvertToSarif(results, version);
+
+                SarifLog log = new SarifLog
+                {
+                    Runs = new Run[] { run }
+                };
+
+                File.WriteAllText(targetFilePath, JsonConvert.SerializeObject(log, Formatting.Indented));
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
+                return false;
+            }
+
+            return true;
+        }
+
+        private List<OwaspMobileTop10Record> ReadFromMd(string folderPath)
+        {
+            List<OwaspMobileTop10Record> records = new List<OwaspMobileTop10Record>();
+            DirectoryInfo d = new DirectoryInfo(folderPath);
+            FileInfo[] files = d.GetFiles("m*-*.md");
+            files = files.OrderBy(f => int.Parse(Regex.Match(f.Name, @"\d+").Value)).ToArray();
+
+            foreach (FileInfo file in files)
+            {
+                List<string> allHeaders = new List<string>();
+                string text = File.ReadAllText(file.FullName);
+                MarkdownDocument doc = Markdig.Markdown.Parse(text);
+                var headerBlocks = doc.Where(d => d.GetType() == typeof(HeadingBlock)).ToList();
+                string titleText = headerBlocks.Select(h => ((LeafBlock)h).Inline.FirstChild.ToString().Trim()).First();
+                Match match = Regex.Match(titleText, "title: \"(?<id>.*): (?<name>.*)\"");
+                var headerTexts = headerBlocks.Select(h => ((LeafBlock)h).Inline.FirstChild.ToString().Trim()).Skip(1).ToList();
+                allHeaders.AddRange(headerTexts);
+                records.Add(new OwaspMobileTop10Record()
+                {
+                    Id = match.Groups["id"].Value,
+                    Name = match.Groups["name"].Value,
+                    ShortDescription = string.Join("\r\n", allHeaders)
+                });
+            }
+
+            return records;
+        }
+
+        private Run ConvertToSarif(List<OwaspMobileTop10Record> records, string version)
+        {
+            IList<ToolComponent> taxonomies = new List<ToolComponent>();
+            ToolComponent toolComponent = new ToolComponent
```

Comment:
>ToolComponent [](http://example.com/codeflow?start=46&length=13)

do we set release date here?

Reply:
there is no release day in the website and I see the field is optional. so I did not set it.